### PR TITLE
[EdgeDB] ProgressReport Workflow schema

### PR DESCRIPTION
**Pull request by atGit2021**
┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/6049737413) by [Unito](https://www.unito.io)

### CODE REVIEW

File: dbschema/project.esdl [hunk 1]
```diff
@@ -214,6 +214,26 @@ module Project {
...
+    `In Progress -> In Review`,
+    `Translation Done`,
+    `Translation Reject`,
+    `Withdraw Review Request`,
+    `In Review -> Needs Translation`,
+    `Review Reject`,
+    `Review Approve`,
+    Publish,
+  >;
+
```
Comment (bryanjnelson) [inline dbschema/project.esdl:236]: I think these make the most sense inside the Workflow Transition type.  They would then be part of module, so could be renamed and reference like `WorkflowTransition.name` & `WorkflowTransition.type`.

File: dbschema/progress-report-workflow-transition.esdl [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module default {
```
Comment (bryanjnelson) [inline dbschema/progress-report-workflow-transition.esdl:1]: I think this is fine as a separate `esdl`, but I would rename to just `workflow-transition.esdl` and have it inside of the `ProgressReport` module.

**Review by bryanjnelson — CHANGES_REQUESTED** [review]

File: dbschema/workflow-transition.esdl [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      readonly := true;
+    };
+  }
+
+  type InternalTransition extending WorkflowTransition {
+    required name: WorkflowTransition::Name;
+    from: array<ProgressReport::Status> {
+      readonly := true;
+    };
+    `notify`: MembersWithRoles;
```
Comment (bryanjnelson) [inline dbschema/workflow-transition.esdl:19]: @CarsonF - We're a little unsure if we've represented this correctly from the application code.  Thoughts?

**Review by bryanjnelson — COMMENTED** [review]
@CarsonF - We're sending this over your way.  I think @atGit2021 has done a great job thus far, but it is possible both of us are off on something here.  Thoughts appreciated.

**Comment by atGit2021** [comment]
@CarsonF I'm not sure how to resolve the error that's causing the CI tests to fail.

File: dbschema/workflow-transition.esdl
Comment (CarsonF) [inline dbschema/workflow-transition.esdl:1]: I'd like to keep the workflow loose & flexible, which is why it's currently in app code. That also allows us to make changes that affect all envs (without data migration).

The only data point that's currently stored in the DB is the transition ID in the workflow event.
While that's not ideal for viewing event log without API, I think we should keep it this way at least until after migration.

File: dbschema/progress-report.esdl [hunk 4]
```diff
@@ -4,3 +4,14 @@ module default {
     overloaded engagement: LanguageEngagement;
```
Comment (CarsonF) [inline dbschema/progress-report.esdl:4]: Need a `workflowEvents` multi link here.

File: dbschema/workflow-event.esdl [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    required at: datetime {
+      readonly := true;
+    };
+    transition: ProgressReport::InternalTransition {
+      readonly := true;
+    };
+    required status: ProgressReport::Status {
+      readonly := true;
+    };
+    required notes: RichText {
```
Comment (CarsonF) [inline dbschema/workflow-event.esdl:15]:
```suggestion
    notes: RichText {
```

File: dbschema/workflow-event.esdl [hunk 6]
```diff
@@ -0,0 +1,20 @@
+module default {
+  type ProgressReportWorkflowEvent {
+    required who: User {
+      readonly := true;
```
Comment (CarsonF) [inline dbschema/workflow-event.esdl:4]:
```suggestion
      readonly := true;
      default := currentUser;
```

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: dbschema/workflow-event.esdl [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    };
+    required at: datetime {
+      readonly := true;
```
Comment (CarsonF) [inline dbschema/workflow-event.esdl:7]:
```suggestion
      readonly := true;
      default := datetime_of_statement();
```

**Review by CarsonF — APPROVED** [review]

**Pull request merged**